src/lib/validations/line-by-line.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @flow
+ * @format
+ */
+
+/**
+ * External dependencies
+ */
+import readline from 'readline';
+import fs from 'fs';
+import chalk from 'chalk';
+import debugLib from 'debug';
+
+/**
+ * Internal dependencies
+ */
+
+const debug = debugLib( 'vip:validations:line-by-line' );
+export type PerLineValidationObject = {
+	execute: Function,
+	postLineExecutionProcessing?: Function,
+};
+
+export type PostLineExecutionProcessingParams = {
+    appId?: number,
+    envId?: number,
+    fileName?: string,
+    isImport?: boolean,
+}
+
+function openFile( filename, flags = 'r', mode = 666 ) {
+	return new Promise( ( resolve, reject ) => {
+		fs.open( filename, flags, mode, ( err, fd ) => {
+			if ( err ) {
+				return reject( err );
+			}
+			resolve( fd );
+		} );
+	} );
+}
+
+export async function getReadInterface( filename: string ) {
+	let fd;
+	try {
+		fd = await openFile( filename );
+	} catch ( e ) {
+		console.log( chalk.red( 'Error: ' ) + 'The file at the provided path is either missing or not readable.' );
+		console.log( 'Please check the input and try again.' );
```

Comment:
Is this an opportunity to use the `exit.withError()` ?

Reply:
I will check this out, but I believe you're correct 👍